### PR TITLE
caddyfile: add support for remaining configs while adding support for placeholder of empty values

### LINKED ISSUE #1: Return `-` when value is not present
I'm using the `format-encoder` module in Caddy v2 to format my logs with a few headers and other fields. I noticed that for example, if I specify the `template` as:

`'{common_log} {request>headers>Referer} {request>headers>User-Agent} {request>host} {resp_headers>X-Request-Id} {duration}'`

Sometimes HTTP requests won't have the `Referer` request header (first access to the page). In those situations, its cleaner (IMO) to return `-` instead of the placeholder.

I've got logs being generated like this:

`127.0.0.1 - - [06/Oct/2020:22:00:23 +0000] "GET /.well-known HTTP/2.0" 302 0 {request>headers>Referer} ["Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:81.0) Gecko/20100101 Firefox/81.0"] localhost ["1d3c101945cc400196a1d5518e7a212f"] 0.00328956`

Same thing when the `{request>headers>User-Agent}` is not present:

`127.0.0.1 - - [06/Oct/2020:22:00:23 +0000] "GET /.well-known HTTP/2.0" 302 0 {request>headers>Referer}  {request>headers>User-Agent} localhost ["76059f3e7b374da2a4e24a80578c3b29"] 0.00267243`

### PR DESCRIPTION
**Pull request by Mohammed90**
Closes #1

### CODE REVIEW

File: caddyfile.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+import (
+	"github.com/caddyserver/caddy/v2/caddyconfig/caddyfile"
+)
+
+// UnmarshalCaddyfile sets up the module from Caddyfile tokens. Syntax:
+//
+//     formatted [<template>] [{
+//			placeholder	[<placeholder>]
+//		}]
```
Comment (francislavoie) [inline caddyfile.go:25]: I think there's a whitespace issue here (tabs -> spaces?)

File: caddyfile.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+// subdirectives that are common to most/all encoders.
+func (se *FormattedEncoder) UnmarshalCaddyfile(d *caddyfile.Dispenser) error {
+	for d.Next() {
+		args := d.RemainingArgs()
+		switch len(args) {
+		case 0:
+			se.Template = commonLogFormat
+		case 1:
+			se.Template = args[0]
+		default:
```
Comment (francislavoie) [inline caddyfile.go:38]: Alternatively you could just join all the remaining args with spaces, would avoid requiring wrapping with quotes here. It's how CEL expressions are handled usually, works ok in simple cases

File: caddyfile.go (hunk 1 above)
Comment (Mohammed90) [reply caddyfile.go]: Yep. Fixed.

File: caddyfile.go (hunk 2 above)
Comment (Mohammed90) [reply caddyfile.go:38]: Done.

**Review by francislavoie — APPROVED** [review]

**Pull request merged**